From 013321eb520d2a12b91d3916c53b76a756bd6475 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 10 Sep 2024 18:51:09 +0300
Subject: [PATCH] [FSW-8694] fs1: manifest based spi efuse init

Initialize spi efuse offset array based on
manifest index. First attempt made with default
index and use second manifest if any failures.

Change-Id: Ic6b9eeab31257b8351f79ac10776c1b0041183bd
---

diff --git a/common/boot_flow.c b/common/boot_flow.c
index 6aaf251..7aee0bd 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -280,3 +280,22 @@
 
 	return desc.spi_offset;
 }
+
+enum boot_rom_status boot_flow_spi_efuse_manifest_init()
+{
+	enum hl_fw_manifest_type curr_manifest_idx;
+
+	/* TODO: primary manifest init should be part of boot rom init */
+	if (!!boot_rom_global.fw_upd_switch_manifest)
+		primary_manifest_idx = HL_FW_SPI_MANIFEST_1;
+
+	curr_manifest_idx = primary_manifest_idx;
+	if (boot_desc_spi_efuse_init(MAX_DIE_COUNT, curr_manifest_idx) == BOOT_ROM_SUCCESS)
+		return BOOT_ROM_SUCCESS;
+
+	curr_manifest_idx = (primary_manifest_idx + 1) % HL_FW_SPI_MANIFEST_MAX_NUM;
+	if (boot_desc_spi_efuse_init(MAX_DIE_COUNT, curr_manifest_idx) == BOOT_ROM_SUCCESS)
+		return BOOT_ROM_SUCCESS;
+
+	return BOOT_ROM_FAIL;
+}
diff --git a/common/boot_flow.h b/common/boot_flow.h
index 9f8554b..f8dc562 100644
--- a/common/boot_flow.h
+++ b/common/boot_flow.h
@@ -21,5 +21,6 @@
 						  uint32_t img_count,
 						  enum boot_rom_progress img_sts_base);
 uint32_t boot_flow_get_spi_efuse_manifest_offset(enum hl_fw_manifest_type manifest_idx);
+enum boot_rom_status boot_flow_spi_efuse_manifest_init();
 
 #endif /* _BOOT_FLOW_H_ */
